plugins/esi/combo_handler.cc

```diff
@@ -585,18 +597,21 @@ getClientRequest(TSHttpTxn txnp, TSMBuffer bufp, TSMLoc hdr_loc, TSMLoc url_loc,
   int         query_len;
   const char *query = TSUrlHttpQueryGet(bufp, url_loc, &query_len);
 
+  // Get URL string for error reporting.
+  std::string url_string = getUrlString(bufp, url_loc);
```

Comment:
In the happy path, this is unused.  Could move this call so it only happens when necessary.